src/Instrumentation/Symfony/src/HttpClientInstrumentation.php

```diff
@@ -0,0 +1,114 @@
+                    ->setAttribute(TraceAttributes::HTTP_METHOD, $params[0])
+                    ->setAttribute(TraceAttributes::CODE_FUNCTION, $function)
+                    ->setAttribute(TraceAttributes::CODE_NAMESPACE, $class)
+                    ->setAttribute(TraceAttributes::CODE_FILEPATH, $filename)
+                    ->setAttribute(TraceAttributes::CODE_LINENO, $lineno);
+
+                $propagator = Globals::propagator();
+                $parent = Context::getCurrent();
+
+                $span = $builder
+                    ->setParent($parent)
+                    ->startSpan();
+
+                if (!isset($params[2]['headers'])) {
+                    $params[2]['headers'] = [];
+                }
+
+                $previousOnProgress = $params[2]['on_progress'] ?? null;
+
+                //As Response are lazy we end span when status code was received
+                $params[2]['on_progress'] = static function (int $dlNow, int $dlSize, array $info) use (
+                    $previousOnProgress,
+                    $span
+                ): void {
+                    if (null !== $previousOnProgress) {
+                        $previousOnProgress($dlNow, $dlSize, $info);
+                    }
+
+                    $statusCode = $info['http_code'];
+
+                    if (0 !== $statusCode && null !== $statusCode && $span->isRecording()) {
+                        $span->setAttribute(TraceAttributes::HTTP_STATUS_CODE, $statusCode);
+
+                        if ($statusCode >= 400 && $statusCode < 600) {
+                            $span->setAttribute(TraceAttributes::HTTP_STATUS_CODE, $statusCode);
+                            $span->setStatus(StatusCode::STATUS_ERROR);
+                        }
+
+                        $span->end();
+                    }
+                };
+
+                $context = $span->storeInContext($parent);
+                $propagator->inject($params[2]['headers'], ArrayAccessGetterSetter::getInstance(), $context);
+
+                Context::storage()->attach($context);
+
+                return $params;
+            },
+            post: static function (
+                HttpClientInterface $client,
+                array $params,
+                ?ResponseInterface $response,
+                ?\Throwable $exception
+            ): void {
+                $scope = Context::storage()->scope();
+                if (null === $scope) {
+                    return;
+                }
+                $scope->detach();
+                $span = Span::fromContext($scope->context());
+
+                if (null !== $exception) {
+                    $span->recordException($exception, [
+                        TraceAttributes::EXCEPTION_ESCAPED => true,
+                    ]);
+                    $span->setStatus(StatusCode::STATUS_ERROR, $exception->getMessage());
+                    $span->end();
+                }
+            },
```

Comment:
Is lack of `$span->end()` at the end intentional?

Reply:
as @brettmc mentioned here https://github.com/open-telemetry/opentelemetry-php-contrib/pull/130#pullrequestreview-1358136402 span is ending in `onprogress` callback because response is lazy and it's not called directly after request method

Reply:
ok, so then I would suggest to add comment here as well to avoid confusions.

Reply:
looks good to me. thx